### ISSUE TITLE
Unable to use preLaunchTask with two configurations

### ISSUE THREAD

**Issue by edvinv** [Microsoft/vscode#24595]
- VSCode Version: 1.11.1
- OS Version: ubuntu

Steps to Reproduce:
1. Create two watching task using typescript compiler named srv and web in tasks.json.
2. In launch.json create three configurations: 
      - srv for node with preLaunchTask=srv 
      - web from chrome with preLaunchTask=web
      - compound launch named both, which starts both launch configurations: srv and web

If I start srv debugging all works ok. But if I then start web debugging I get "There is a task srv running. Can not run pre launch task web." Same happens if I start compound task named both.

But I'm able to run both tasks srv and web at the same time if I start them with Run task command. Of course, I must remove preLaunchTask, if I want to start debugging, otherwise I get same error.

br

**Comment by isidorn** [comment]
This is the problem of the task framework only supporting to run one task at a time.
Assigning to @dbaeumer as he might have better recomendations on how to setup your environment

**Comment by edvinv** [comment]
@isidorn not sure, because I use tasks version 2.0.0. and I'm able to run both watch task at the same time without problems. But if two watch task are running I'm unable to start debugging if preLaunchTask is also set.

**Comment by isidorn** [comment]
@edvinv this issue is probably that the `preLaunchTask` uses tasks 1.0.0 underneath

**Comment by dbaeumer** [comment]
@isidorn  actually the new runner support starting more than one tasks but the debugger needs to adopt this. Since the new terminal is not the default yet I recommend that we wait with the adoption. What is clear is that I need to let the debugger now whether the task framework supports multiple parallel tasks or not since users could still configure the old runner. Opened https://github.com/Microsoft/vscode/issues/24670 to track this.

Assigning back to you.

**Comment by isidorn** [comment]
Assigning to Mai since this is dependent on #24670

**Comment by isidorn** [comment]
@dbaeumer let me know if you are not planning to tackl #24670 in May so I also postpone this

**Comment by dbaeumer** [comment]
@isidorn sorry for this. Lets move to June. It is a little late for feature although this is not too complicated.

**Comment by edvinv** [comment]
@isidorn @dbaeumer was this corrected in last release 1.18? Because now I'm able to have compound launch configuration, containing two launch configurations,  each having different preLaunchTask task and Start debugging (F5) correctly starts both prelaunch  tasks.

The problem is then is that debugger doesn't wait for both tasks to end before attaching debug session. 
If I start  single webpack prelaunch tasks I also get error "The preLaunchTask 'web' cannot be tracked."

**Comment by dbaeumer** [comment]
This got addressed with the new task 2.0.0 support which allows to run multiple launch configs. The canot be tracked issues is discussed here: #37997
